responsibleai/responsibleai/rai_insights/rai_insights.py

```diff
@@ -489,6 +489,23 @@ def _write_to_file(self, file_path, content):
         with open(file_path, 'w') as file:
             file.write(content)
 
+    def _reset_data_frame_index(self, some_dataframe):
+        """Reset the index if duplicates exist in the dataframe index.
+           Create a new dataframe if the input dataframe has duplicate index.
+
+        :param some_dataframe: Dataframe to check for duplicate index
+        :type some_dataframe: pd.DataFrame
+        :return: Dataframe without duplicate indices.
+        :rtype: pd.DataFrame
+        """
+        num_indices = len(some_dataframe.index.tolist())
+        num_distinct_indices = len(np.unique(some_dataframe.index).tolist())
+
+        if num_indices == num_distinct_indices:
+            return some_dataframe
+
+        return some_dataframe.reset_index(drop=True)
```

Comment:
I don't think we can drop the index like this:
```drop=True```
I recall there were many cases where this is needed by the learner itself to make predictions, especially in time series scenarios?

Reply:
The reset_index call doesn't drop the index but resets it to be uniue. Documentation is here https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.reset_index.html.

The timeseries models should have unique indexes right?

Reply:
discussed offline - when drop=True, the index is reset to default monotonically increasing values, which is effectively dropping that information.  However, there are some weird models/scenarios we've encountered in past where this index is actually used.  There may also be tricky cases with MultiIndex scenarios.  If possible, the ideal solution would be to make the index a regular column through reset_index without drop=True, and then call set_index on deserialize to rehydrate the index column and recreate the exact same dataset as the original one.

Reply:
the option orient='split' in to_json() saves index as a separate entity in the json file so even if duplicate index exists in the dataframe.